grug/math/src/decimal.rs

```diff
@@ -11,27 +11,168 @@ pub trait Decimal: Sized {
 impl<U> Decimal for Dec<U>
 where
     Self: FixedPoint<U>,
-    U: Number + Copy + PartialEq,
+    U: Number + Copy + PartialEq + Sign + NumberConst,
```

Comment:
Let's order the trait bounds such that all number traits come first, then all the std traits